cypress/tests/ui/toolbar/global-toolbar.cy.js

```diff
@@ -1 +1,160 @@
+      .should('be.visible')
+      .then(($dialog) => {
+        cy.wrap($dialog)
+          .should('have.attr', 'class')
+          .and('contains', 'modal--visible');
+
+        cy.wrap($dialog).within(() => {
+          cy.get('.modal__title').should('have.text', 'Settings');
+        });
+      });
+  });
+
+  describe('Settings Panel', () => {
+    it('verifies that users can change the display of the names of their kedro nodes. #TC-5', function () {
+      // Alias
+      cy.get('[data-test="pipeline-toggle-input-isPrettyName"]').as(
+        'isPrettyNameCheckbox'
+      );
+
+      // Assert before action
+      cy.get('@isPrettyNameCheckbox').should('be.checked');
+
+      // Check if the pipeline name is pretty
+      cy.get(':nth-child(1) > .menu-option__content')
+        .should('have.attr', 'title')
+        .should('not.match', /[_-]/);
+
+      // Check if the metadata panel node name label matches
+      cy.get('.pipeline-node__bg').first().click({ force: true });
+      cy.get('.pipeline-metadata__properties > :nth-child(1)').should(
+        'have.text',
+        'Original node name:'
+      );
+
+      // Action
+      cy.get('@isPrettyNameCheckbox').uncheck({ force: true });
+      cy.get('[data-test="Apply changes and close in Settings Modal"]').click({
+        force: true,
+      });
+
+      // Assert after action
+      cy.__waitForPageLoad__(() => {
+        // Check if the pipeline name is original
+        cy.get(':nth-child(1) > .menu-option__content')
+          .should('have.attr', 'title')
+          .should('match', /[_-]/);
+
+        // Check if the metadata panel node name label matches
+        cy.get('.pipeline-node__bg').first().click();
+        cy.get('.pipeline-metadata__properties > :nth-child(1)').should(
+          'have.text',
+          'Pretty node name:'
+        );
+      });
+    });
+
+    it('verifies that users can show a warning before rendering very large graphs. #TC-6', function () {
+      // Alias
+      cy.get('[data-test="pipeline-toggle-input-sizewarning"]').as(
+        'isSizeWarning'
+      );
+
+      // Assert before action
+      cy.get('@isSizeWarning').should('be.checked');
+
+      // Intercept the network request to mock with a fixture
+      cy.__interceptRest__('/api/main', 'GET', '/mock/largeDataset.json');
+      cy.reload();
+
+      cy.__waitForPageLoad__(() => {
+        cy.get('.pipeline-warning__title')
+          .should('exist')
+          .and('have.text', 'Whoa, that’s a chonky pipeline!');
+      });
+
+      // Action
+      cy.get('@isSizeWarning').uncheck({ force: true });
+      cy.get('[data-test="Apply changes and close in Settings Modal"]').click({
+        force: true,
+      });
+
+      // Assert after action
+      cy.__waitForPageLoad__(() => {
+        cy.get('.pipeline-warning__title').should('not.exist');
+      });
+    });
+
+    it('verifies that users can expand all modular pipelines on first load. #TC-7', function () {
```

Comment:
in this should we check if a modular pipeline child is first not shown in the graph and then it is shown when everything is expanded maybe?